lib/publication.js

```diff
@@ -90,18 +89,38 @@ class Publication {
   }
 
   _publishChildrenOf (doc) {
-    const children = _.isFunction(this.childrenOptions)
+    const children = typeof this.childrenOptions === 'function'
       ? this.childrenOptions(doc, ...this.args) : this.childrenOptions
-    _.each(children, function createChildPublication (options) {
+    children.forEach((options) => {
       const pub = new Publication(this.subscription, options, [doc].concat(this.args))
       this.publishedDocs.addChildPub(doc._id, pub)
       pub.publish()
-    }, this)
+    })
   }
 
   _republishChildrenOf (doc) {
+    const parentArgs = this.args
+    let newArgs
     this.publishedDocs.eachChildPub(doc._id, (publication) => {
+      // First argument is the new document
       publication.args[0] = doc
+
+      // Check if parent's args are the same length as this publication
+      // Intuitively this should not ever be the case! However it does happen sometimes.
+      // When it does the first argument of the parent publication is the doc.
+      // So we skip this to avoid creating a duplicate of the first argument.
+      if (parentArgs.length === publication.args.length) {
+        newArgs = parentArgs.slice(1)
+      } else {
+        newArgs = parentArgs
+      }
+
+      // Subsequent args are passed down from parent.
+      // These may have been updated by a grandparent publication.
+      newArgs.forEach(function (arg, i) {
+        publication.args[i + 1] = arg
+      })
```

Comment:
could we get rid of line 106 and replace this forEach with:

```javascript
publication.args = [doc, ...newArgs];
```

Reply:
Yes!